sycl/doc/extensions/proposed/sycl_ext_oneapi_usm_malloc_properties.asciidoc

```diff
@@ -0,0 +1,1319 @@
+annotated_ptr<void, propertyListB>
+malloc_annotated(
+   size_t numBytes,
+   const queue& syclQueue,
+   const propertyListA &propList)
+----
+a@ Simplified form where `syclQueue` provides the `context`
+and any necessary `device`.
+
+a@
+[source]
+----
+template <typename T,
+          typename propertyListA,
+          typename propertyListB>
+annotated_ptr<T, propertyListB>
+malloc_annotated(
+   size_t count,
+   const queue& syclQueue,
+   const propertyListA &propList)
+----
+a@ Simplified form where `syclQueue` provides the `context`
+and any necessary `device`.
+
+|====
+
+=== USM Memory Allocation Properties
+
+This section specifies the properties that can be passed to the USM memory allocation functions with `properties` support defined in Section <<section.usm.allocs>>.
+A SYCL implementation or SYCL backend may support additional properties other than those defined here, provided they are defined in accordance with the
+link:../experimental/sycl_ext_oneapi_properties.asciidoc[sycl_ext_oneapi_properties] extension.
+If unsupported properties are passed to the USM memory allocation functions with `properties` support, the compiler will issue a diagnostic error.
+
+All properties supported by `annotated_ptr` are supported by the USM memory allocation functions with `properties` support. These properties do not affect the behavior of the allocation functions at runtime, except where explicitly specified.
+
+
+[NOTE]
+====
+The USM memory allocation functions with `properties` support defined in this extension support all properties supported by `annotated_ptr`, as a convenience. This allows the user to attach all desired properties to the `annotated_ptr` in a single step.
+====
+
+Table <<table.usm.malloc.properties>> lists properties that affect the behavior of the allocation functions.
+
+[[table.usm.malloc.properties]]
+.Properties that affect the behavior of the USM memory allocation functions with properties support
+[options="header"]
+|====
+|Property|Description|Source
+a|
+sycl::ext::oneapi::experimental::alignment
+| 	
+If this property is passed to a USM memory allocation function with `properties` support, it instructs the runtime to allocate memory with this alignment in bytes.
+The set of allowed alignments is implementation defined. Specifying an alignment that is not supported causes the allocation function to return an `annotated_ptr` containing a raw pointer that is `nullptr`.
+|
+link:../proposed/sycl_ext_oneapi_annotated_ptr.asciidoc[sycl_ext_oneapi_annotated_ptr]
+|====
+
+Table <<table.usm.malloc.devhostsh>> lists the new properties introduced by this extension.
+
+[[table.usm.malloc.devhostsh]]
+.New compile-time constant properties introduced by this extension
+[options="header"]
+|====
+|Property|Values|Description|Supported By
+a|
+sycl::ext::oneapi::experimental::usm_kind
+| 	
+sycl::usm::alloc::host
+sycl::usm::alloc::device
+sycl::usm::alloc::shared
```

Comment:
We should somehow document all the different variants of this property.  For an "enum" property, we usually define a shortcut property name for each enumeration value.  There is also a "_key" for each property.  Other specs show this via a code synopsis (e.g. [sycl_ext_intel_fpga_kernel_arg_properties](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_intel_fpga_kernel_arg_properties.asciidoc)).

I think the properties should be:

* `template<sycl::usm::alloc A> usm_kind`
* `usm_kind_host`
* `usm_kind_device`
* `usm_kind_shared`

And then there is also a "key" type used for queries `usm_kind_key`.

Reply:
Ok, I've made these changes. Thanks!